### PR TITLE
Typo in docs + some stylistic changes

### PR DESCRIPTION
**Pull request by xkollar**
Hello there,

This PR is mostly motivated by the typo in docs, but once I had things checked out I had a bit of looksies around and have couple of suggestions, mostly for stylistic consistency.

There is one sus thing that I have marked with `!!!`, which might be okay, but after rewriting the condition to avoid negation it seems a bit suspicious... person familiar with code base would likely be able to provide insight into whether it is okay/intended or whether that is a real problem/bug.

### CODE REVIEW

File: src/Data/DecisionDiagram/BDD.hs [hunk 1]
```diff
@@ -391,8 +390,9 @@ pbAtLeast xs k0 = unfoldOrd f (0, k0)
 
     f :: (Int, w) -> Sig (Int, w)
     f (!i, !k)
-      | not (k <= ub) = SLeaf False
+      | k > ub = SLeaf False
       | i == V.length xs' && 0 >= k = SLeaf True
+      -- !!! SUS?
```
Comment (xkollar) [inline src/Data/DecisionDiagram/BDD.hs:395]: The sus part

File: src/Data/DecisionDiagram/BDD.hs [hunk 2]
```diff
@@ -391,8 +390,9 @@ pbAtLeast xs k0 = unfoldOrd f (0, k0)
 
     f :: (Int, w) -> Sig (Int, w)
     f (!i, !k)
```
Comment (xkollar) [inline src/Data/DecisionDiagram/BDD.hs:392]: The suspicious part

**Comment by msakai** [comment]
Thank you for the PR.
I'll take a look.

**Comment by xkollar** [comment]
Thank you. I'm also open to suggestion as to how to change the MR...

File: src/Data/DecisionDiagram/BDD.hs (hunk 1 above)
Comment (msakai) [inline src/Data/DecisionDiagram/BDD.hs:395]: The function `f` encodes inequality constraint `wi*xi + … + wn*xn ≥ k`.

Here `ub := max_{xi,…,xn} wi*xi + … + wn*xn` and `lb := min_{xi,…,xn} wi*xi + … + wn*xn` .

* In the clause above the comment, the condition `i == V.length xs' && 0 >= k` means that the left-hand side `wi*xi + … + wn*xn` is empty and the right-hand side `k` is less than or equal to 0. Therefore `wi*xi + … + wn*xn ≥ k` is true.

* In the clause below the comment, the condition `lb ≥ k` implies`wi*xi + … + wn*xn ≥ lb ≥ k`. Therefore `wi*xi + … + wn*xn ≥ k` is also true.

Now I realized that the former condition `i == V.length xs' && 0 >= k` implies the latter condition `lb ≥ k`, so the former clause is redundant. Similarly, `pbAtMost` function below also has the redundant clause `i == V.length xs' && 0 <= k = SLeaf True`.

If this explains what you suspected, please add a commit to delete those two clauses (`      | i == V.length xs' && 0 >= k = SLeaf True` line above and `      | i == V.length xs' && 0 <= k = SLeaf True` line in the below `pbAtMost`) and the comment line `      -- !!! SUS?`.

Comment (xkollar) [reply src/Data/DecisionDiagram/BDD.hs:395]: Returning to it, I think the "issue" I saw was different (I thought there was difference between equivalent cases of BDD and ZDD, but I checked again and they were the same.

However I looked into what you wrote and it makes sense/is true. To double check I tried running tests with commented out those two clauses. So I'll obliterate the comment and clauses and clean up the commits.

**Comment by xkollar** [comment]
Sorry, I messed up with git, let me clean that up

**Comment by xkollar** [comment]
Now it should be okay, sorry for the troble. Also, in case you don't want the `.nix` file in your repo I can drop the commit... it makes it easier for me to develop on things but I know it is not everyone's cup of tea.

**Comment by msakai** [comment]
I'm not familiar with Nix, but I guess that adding `.nix` file is convenient for those using nix and does not interfere with others.
Then I think it's okay to add.

**Review by msakai — APPROVED** [review]
Thank you for catching so many issues.
It looks good to me.
(Though, I have no strong opinion on which style is better, before or after the changes, in many cases.)

**Pull request merged**